src/app/modules/entity/entity-form/validators/forbidden-values-validation/forbidden-values-validation.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  AsyncValidatorFn, FormControl, ValidationErrors, ValidatorFn,
+} from '@angular/forms';
+import { map, Observable, take } from 'rxjs';
+
+export function forbiddenValues(arrayOfValues: unknown[], caseInsensitive?: boolean): ValidatorFn {
+  let thisControl: FormControl<string>;
+
+  return (control: FormControl): ValidationErrors | null => {
+    if (!control.parent) {
+      return null;
+    }
+
+    // Initializing the validator.
+    if (!thisControl) {
```

Comment:
As discussed on the call I don't know why we are checking for parent or caching control.